src/libraries/System.Text.RegularExpressions/tests/RegexSRMTests.cs

```diff
@@ -91,72 +91,205 @@ public void TestStartSet()
+        //var re = new Regex(@"\b\d{1,2}\/\d{1,2}\/\d{2,4}\b", DFA | RegexOptions.Singleline);
+        //var re = new Regex(@"(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9])", DFA | RegexOptions.Singleline);
+        //var re = new Regex(@"[\w]+://[^/\s?#]+[^\s?#]+(?:\?[^\s#]*)?(?:#[^\s]*)?", DFA | RegexOptions.Singleline);
+        //var re = new Regex(@"\b\w+nn\b", DFA);
+        //var re = new Regex(@"abracadabra\z", DFA | RegexOptions.Multiline);
+        //var re = new Regex("^.{5}", DFA);
+        //var re = new Regex("^a(_?a?_?a?_?)+$", DFA);
+        //var re = new Regex(@"^\b[a-z]+(n|\n|@|_)n\b", DFA);
+        //var re = new Regex(@"^[a-z]+", DFA | RegexOptions.Multiline);
+        //var re = new Regex(@"(([a-z])+.)+[A-Z]([a-z])+", DFA | RegexOptions.Singleline);
+        //var re = new Regex("ab+c$", DFA);
+        //ViewDGML(regex: re);
+        //ViewDGML(regex: re, addDotStar: true);
+        //ViewDGML(regex: re, inReverse: true);
+        //var match = re.Match("_.@[0.0.0.aaaaaaaaaa]");
+        //string[] rawregexes = File.ReadAllLines("c:/tmp/VS19preview/cortana.txt");
+        //HashSet<int> notsupported = new();
+        //HashSet<string> set = new();
+        //Regex QsizeMatcher = new Regex(@"\|Q\|=[0-9]+", DFA);
+        //Regex DsizeMatcher = new Regex(@"\|&#x0394;\|=[0-9]+", DFA);
+        //Regex AsizeMatcher = new Regex(@"\|&#x03A3;\|=[0-9]+", DFA);
+        //Func<string, int> NrOfStates = (s) => int.Parse(QsizeMatcher.Match(s).Value.Substring(4));
+        //Func<string, int> NrOfMoves = (s) => int.Parse(DsizeMatcher.Match(s).Value.Substring(11));
+        //Func<string, int> NrOfSymbols = (s) => int.Parse(AsizeMatcher.Match(s).Value.Substring(11));
+        //HashSet<int> stateSizes = new();
+        //HashSet<int> moveSizes = new();
+        //HashSet<int> alphSizes = new();
+        //for (int i = 0; i < rawregexes.Length; i++)
+        //{
+        //    var re = new Regex(rawregexes[i], DFA);
+        //    StringWriter sw = new StringWriter();
+        //    SaveDGML(re, sw, onlyDFAinfo: true);
+        //    ViewDGML(re);
+        //    int n = NrOfStates(sw.ToString());
+        //    int m = NrOfMoves(sw.ToString());
+        //    int k = NrOfSymbols(sw.ToString());
+        //    stateSizes.Add(n);
+        //    moveSizes.Add(m);
+        //    alphSizes.Add(k);
+        //}
+        //Console.WriteLine(stateSizes.Count);
+        //string rawregex = @"((decryptionKey|validationKey)=""[a-zA-Z0-9]+"")";
+        //string rawregex = @"(?<Part0>(\b(10|11|12|[1-9])\b))\s*(?<Part1>(:)\s*(?<Part2>([0-5][0-9]))){0,1}\s*(?<Part3>(\b([Tt]he day before yesterday|day before yesterday|[Tt]he day after tomorrow|day after tomorrow|[Yy]esterday|[Tt]oday|[Tt]ommorrow|[Tt]omorrow|[Tt]ommorow|[Tt]mrw)\b))";
+        //string rawregex = @"(?<Part0>(tonight))\s*(?<Part1>(at)){0,1}\s*(?<Part2>(midnight))";
+        */
+
+        private const string inputfile = @"c:\tmp\vsinput.txt";
```

Comment:
For temporary files, we make the test class derive from FileCleanupTestBase. Then you call GetTestFileName() one or more times. It will create a random name, in an appropriate path, and take care of deleting them after you tests are done.

Reply:
These are temporary changes that I intend to delete later, they are not really used in any of the regular unit tests but as  input/output files is a couple of lightweight profiling entry points for me in couple of unit tests that are actually commented out as unit tests when I do the PR (actually two methods: called TestRunPerformace and TestRunSerialization).
For temporary files I've usually been using in-memory StringWriter, and avoiding actual file creation (eg in the unit test that tests DGML generation).